### ISSUE TITLE
Consider adding an "index type" field next to the index definitions in listIndexes response

### ISSUE THREAD

**Issue by hemidactylus** [stargate/data-api#1696]
Readers have to rely on the presence of specific fields within `definition` to figure out if an index is vector, "regular" (and in the future there'll be more types).
The suggestion is also to mitigate the asymmetry with the various createXYZIndex commands and add a new field in the `listIndexes` response, such as:

```
{
    "status": {
        "indexes": [
            {
                "name": "idx_table_find_p_int",
                "indexType": "justVanilla",        <===== THIS!
                "definition": {
                    "column": "p_int",
                    "options": {}
                }
            },
            {
                "name": "idx_table_find_p_vector",
                "indexType": "vector",             <===== THIS!
                "definition": {
                    "column": "p_vector",
                    "options": {
                        "metric": "cosine",
                        "sourceModel": "other"
                    }
                }
            }
        ]
    }
}
```

Note this is not inside "definition", to preserve identity between what gets in (createIndex) and what goes out (listIndexes)

**Comment by vkarpov15** [comment]
I like the general idea of being able to create a vector index using `createIndex`, without need for separate `createVectorIndex`. But 2 suggestions:

1. I don't like `justVanilla` as the default `indexType`
2. Would it be possible to simply infer that an index is a vector index if and only if `options.metric` is specified?

**Comment by hemidactylus** [comment]
I agree, indeed "justVanilla" was never meant to be a serious suggestion, just a placeholder!

Also, yes, it is currently the way us (clients) do when marshaling the response into named objects (instances of TableVectorIndex or TableIndex respectively (can't remember the exact names right now). But my point is that such an indirect way to discern whether vector or not is not very robust. Things may (and will) change, new index types will be introduced, etc.

**Comment by amorton** [comment]
the fix is: 

- add `indexType` to the response of the listIndexes command, add this to the `io.stargate.sgv2.jsonapi.api.model.command.table.IndexDesc` interface and implement as needed 
- all createindex to accept an optional `indexType` field that must be of the appropriate type from the enum above (if specified) check this in the `CreateIndexCommandResolver` or `CreateIndexAttemptBuilder`
-

**Comment by amorton** [comment]
Re-opening - the merged code failed to remove the "collection" index type per the ticket.
